core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -629,12 +630,15 @@ class KafkaApis(val requestChannel: RequestChannel,
       AdminUtils.createTopic(zkUtils, topic, numPartitions, replicationFactor, properties, RackAwareMode.Safe)
       info("Auto creation of topic %s with %d partitions and replication factor %d is successful"
         .format(topic, numPartitions, replicationFactor))
-      new MetadataResponse.TopicMetadata(Errors.LEADER_NOT_AVAILABLE, topic, java.util.Collections.emptyList())
+      new MetadataResponse.TopicMetadata(Errors.LEADER_NOT_AVAILABLE, topic, common.Topic.isInternal(topic),
```

Comment:
I'm wondering if we can push down the work of figuring out if a topic is internal to TopicMetadata constructor. It will clean up the API and TopicMetadata has all the information it needs to find out.